src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js

```diff
@@ -235,7 +235,7 @@ const getSaveHandler = (
             } = this.props;
 
             const filteredProps = onFilterProps ? onFilterProps(passOnProps) : passOnProps;
-            this.isCompleting = !!(onIsCompleting && onIsCompleting(this.props));
+            this.isCompleting = Boolean(onIsCompleting && onIsCompleting(this.props) === 'true');
```

Comment:
Can we change the edit event code instead of the generic withSaveHandler component? here: https://github.com/dhis2/capture-app/blob/dbc105106c3a2efc452fb11bdcd85a831de3821e/src/core_modules/capture-core/components/WidgetEventEdit/EditEventDataEntry/EditEventDataEntry.component.js#L360

I think it makes sense to keep the return value of onIsCompleting a boolean value.